hyperspy/tests/axes/test_data_axis.py

```diff
@@ -314,6 +314,46 @@ def test_calibrate(self):
         with pytest.raises(TypeError, match="only for uniform axes"):
             self.axis.calibrate(value_tuple=(11,12), index_tuple=(0,5))
 
+    def test_functional_value2index(self):
```

Comment:
Because this test is for a method of `BaseDataAxis`, it would be better to move this test to the `TestBaseDataAxis` class.

Reply:
I thought too, but since BaseDataAxis can practically not be instantiated it felt awkward. It is, indeed, a method of BaseDataAxis but will never ever be called outside the subclasses. 

I can still move it there if you think that's better. I'll need to do something like 
```python
bax = BaseDataAxis()
bax.size = 100
etc...
```

Reply:
I would say that it would still be more tidy to have this test in `TestBaseDataAxis` but at the same time, this is true that most test are in `TestDataAxis`!
I was suggesting this because it happen to me in the past to write tests which were already written somewhere else, just not where I expected them to be...

Reply:
Couldn't we add a comment to `TestBaseDataAxis` to indicate that this is tested in `TestDataAxis` @ericpre ?

Reply:
Actually, it is checked for all three axes types individually, so that could be mentioned in the `TestBaseDataAxis` class.

Reply:
>Actually, it is checked for all three axes types individually, so that could be mentioned in the TestBaseDataAxis class.

Ok i will do it. 

I would suggest, in a later PR, to formalize the fact that BaseDataAxis is an abstract class, and provide guidelines for the implementation of different axes types.